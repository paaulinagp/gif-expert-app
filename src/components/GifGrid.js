import React from 'react'

export const GifGrid = ({category}) => {

  const getGifs = async () => {
    const url =`https://api.giphy.com/v1/gifs/search?api_key=4BnvcbsxEOoVInUE8HhNpT7NxafDQy1V&q=Harry Potter`;
    const resp = await fetch(url);
    const { data } = await resp.json();
    const gifs = data.map( (gif) => ({
      id: gif.id,
      title: gif.title,
      img: gif.images?.downsized_medium.url
    }) );
    console.log(gifs)
  };
  getGifs();
  return (
    <div>
      <h3>{category}</h3>
    </div>
  )
}
