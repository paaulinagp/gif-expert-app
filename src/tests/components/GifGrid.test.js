import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs'); // Se mockean los datos del hook

describe('Pruebas unitarias de componente <GifGrid />', () => {
  let wrapper;
  const category = "Category";

  test('debe cargar correctamente el componente ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuanso se cargan las imagenes', () => {
    useFetchGifs.mockReturnValue({
      data: [{
        id: 'ABC',
        title: 'Titulo',
        url: 'https://dato.com'
      }],
      loading: false
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
})
