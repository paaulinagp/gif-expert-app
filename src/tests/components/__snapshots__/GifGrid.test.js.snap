// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias de componente <GifGrid /> debe cargar correctamente el componente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    Category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas unitarias de componente <GifGrid /> debe de mostrar items cuanso se cargan las imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-2s"
  >
    Category
  </h3>
  <div
    className="card-grid"
  >
    <Component
      id="ABC"
      key="ABC"
      title="Titulo"
      url="https://dato.com"
    />
  </div>
</Fragment>
`;
